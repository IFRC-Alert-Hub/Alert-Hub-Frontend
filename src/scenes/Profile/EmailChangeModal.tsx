import { Box, Button, InputAdornment, TextField } from "@mui/material";
import { useEffect, useState } from "react";
import { useFormik } from "formik";
import * as yup from "yup";
import MessageModal from "./components/MessageModal";
import ModalButton from "./components/ModalButton";

type PropsType = {
  isVerified: boolean;
  setIsVerified: React.Dispatch<React.SetStateAction<boolean>>;
};

const EmailChangeModal = ({ isVerified, setIsVerified }: PropsType) => {
  const [isCodeSent, setIsCodeSent] = useState(false);
  const [timeLeft, setTimeLeft] = useState(60);

  useEffect(() => {
    if (isCodeSent) {
      const timer = setInterval(() => {
        setTimeLeft((prevTime) => {
          if (prevTime > 0) {
            return prevTime - 1;
          } else {
            setIsCodeSent(false);
            clearInterval(timer);
            return 60;
          }
        });
      }, 1000);

      return () => {
        clearInterval(timer);
      };
    }
  }, [isCodeSent]);

  const formik = useFormik({
    initialValues: {
      newEmail: "",
      verificationCode: "",
    },
    validationSchema: yup.object({
      newEmail: yup
        .string()
        .email("Invalid email address")
        .required("Please enter the email"),
      verificationCode: yup.string().required("Please enter the code"),
    }),
    onSubmit: () => {
      console.log("Link new Email");
    },
  });

  const handleClose = () => setIsVerified(false);

  const handleCodeSent = () => {
    setIsCodeSent(true);
  };

  return (
    <MessageModal
      header="Modify Email"
      open={isVerified}
      handleClose={handleClose}
    >
      <Box component="form">
        <TextField
          fullWidth
          id="newEmail"
          placeholder="New Email"
          size="small"
          value={formik.values.newEmail}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={formik.touched.newEmail && Boolean(formik.errors.newEmail)}
          helperText={formik.touched.newEmail && formik.errors.newEmail}
        />
        <TextField
          fullWidth
          id="verificationCode"
          placeholder="Verification Code"
          size="small"
          disabled={!formik.touched.newEmail || Boolean(formik.errors.newEmail)}
          value={formik.values.verificationCode}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={
            formik.touched.verificationCode &&
            Boolean(formik.errors.verificationCode)
          }
          helperText={
            formik.touched.verificationCode && formik.errors.verificationCode
          }
          sx={{
            mt: "2rem",
            mb: "2rem",
            "& .MuiOutlinedInput-root": {
              paddingRight: 0,
            },
          }}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <Box borderLeft={"1px solid #ccc"}>
                  <Button
                    variant="text"
                    disableRipple
                    disabled={
                      !formik.touched.newEmail ||
                      Boolean(formik.errors.newEmail) ||
                      isCodeSent
                    }
                    sx={{
                      color: "#d30210",
                      p: "0px 10px 0px 10px",
                      minWidth: "50px",
                      textTransform: "capitalize",
                      "&:hover": { opacity: "0.5" },
                    }}
                    onClick={handleCodeSent}
                  >
                    {isCodeSent && timeLeft > 0 ? (
                      <span>{timeLeft}s</span>
                    ) : (
                      <span>Send</span>
                    )}
                  </Button>
                </Box>
              </InputAdornment>
            ),
          }}
        />
        <ModalButton name="Link" />
      </Box>
    </MessageModal>
  );
};

export default EmailChangeModal;
