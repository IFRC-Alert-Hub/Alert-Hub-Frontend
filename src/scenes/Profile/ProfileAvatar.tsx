import { Avatar, Box, Button } from "@mui/material";

interface User {
  id: string;
  avatar: string;
  firstName: string;
  lastName: string;
  country: string;
  city: string;
  email: string;
  phoneNumber: string;
}

type PropsType = {
  user: User | null;
  editStatus: boolean;
  setEditStatus: React.Dispatch<React.SetStateAction<boolean>>;
};

const ProfileAvatar = ({ user, editStatus, setEditStatus }: PropsType) => {
  const handleEdit = () => {
    setEditStatus(false);
  };

  return (
    <Box
      marginTop="30px"
      display="flex"
      justifyContent="center"
      alignItems="center"
      flexDirection="column"
    >
      <Avatar alt="User Name" src={user?.avatar} className="avatar-size" />
      {editStatus ? (
        <Button variant="text" className="edit-btn" onClick={handleEdit}>
          Edit Profile
        </Button>
      ) : (
        <div></div>
      )}
    </Box>
  );
};

export default ProfileAvatar;
