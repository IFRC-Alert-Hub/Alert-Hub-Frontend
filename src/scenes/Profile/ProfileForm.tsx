import {
  Box,
  Button,
  Grid,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  TextField,
} from "@mui/material";
import { useState } from "react";

interface User {
  id: string;
  avatar: string;
  firstName: string;
  lastName: string;
  country: string;
  city: string;
  email: string;
  phoneNumber: string;
}

type PropsType = {
  user: User | null;
  setUser: React.Dispatch<React.SetStateAction<User | null>>;
  editStatus: boolean;
  setEditStatus: React.Dispatch<React.SetStateAction<boolean>>;
};

const ProfileForm = ({
  user,
  setUser,
  editStatus,
  setEditStatus,
}: PropsType) => {
  const [, setPrevUser] = useState(user);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setUser((prevUser: User | null) => ({
      ...(prevUser as User), // Type assertion to satisfy TypeScript
      [e.target.name]: e.target.value,
    }));
  };
  const handleCancel = () => {
    setUser((prevUser) => prevUser); // No changes needed, just reset to the previous user state
    setEditStatus(true);
  };

  const handleSubmit = () => {
    setPrevUser(user);
    console.log(user);
    setEditStatus(true);
  };

  return (
    <Box
      component="form"
      noValidate
      autoComplete="off"
      className="form-box"
      onSubmit={(e) => e.preventDefault()}
    >
      <Grid container>
        <Grid item xs={12} sm={6}>
          <InputLabel className="form-label" htmlFor="firstName">
            First Name
          </InputLabel>
          <TextField
            id="firstName"
            name="firstName"
            size="small"
            className="form-text-field"
            value={user?.firstName}
            disabled={editStatus}
            onChange={handleInputChange}
          />
        </Grid>
        <Grid item xs={12} sm={6}>
          <InputLabel className="form-label" htmlFor="lastName">
            Last Name
          </InputLabel>
          <TextField
            id="lastName"
            name="lastName"
            size="small"
            className="form-text-field"
            value={user?.lastName}
            disabled={editStatus}
            onChange={handleInputChange}
          />
        </Grid>
        <Grid item xs={12} sm={6}>
          <InputLabel className="form-label" htmlFor="country">
            Country
          </InputLabel>
          <TextField
            id="country"
            name="country"
            size="small"
            className="form-text-field"
            value={user?.country}
            disabled={editStatus}
            onChange={handleInputChange}
          />
        </Grid>
        <Grid item xs={12} sm={6}>
          <InputLabel className="form-label" htmlFor="city">
            City
          </InputLabel>
          <TextField
            id="city"
            name="city"
            size="small"
            className="form-text-field"
            value={user?.city}
            disabled={editStatus}
            onChange={handleInputChange}
          />
        </Grid>

        <Grid item xs={12} m={{ xs: "0px 0px 20px 0px", sm: "0 30px 20px 0" }}>
          <InputLabel htmlFor="email">Email Address</InputLabel>
          <OutlinedInput
            id="email"
            name="email"
            size="small"
            value={user?.email}
            disabled
            fullWidth
            sx={{ p: "0px" }}
            endAdornment={
              <InputAdornment position="end">
                <Button
                  variant="text"
                  disabled={!editStatus}
                  disableRipple
                  sx={{
                    color: "red",
                    textTransform: "capitalize",
                    "&:hover": { opacity: "0.5" },
                  }}
                >
                  {"Switch"}
                </Button>
              </InputAdornment>
            }
          />
        </Grid>
        <Grid item xs={12} m={{ xs: "0px 0px 20px 0px", sm: "0 30px 30px 0" }}>
          <InputLabel htmlFor="phoneNumber">Phone Number</InputLabel>
          <OutlinedInput
            id="phoneNumber"
            name="phoneNumber"
            size="small"
            value={user?.phoneNumber}
            disabled
            fullWidth
            sx={{ p: "0px" }}
            endAdornment={
              <InputAdornment position="end">
                <Button
                  variant="text"
                  disabled={!editStatus}
                  disableRipple
                  sx={{
                    color: "red",
                    textTransform: "capitalize",
                    "&:hover": { opacity: "0.5" },
                  }}
                >
                  {"Switch"}
                </Button>
              </InputAdornment>
            }
          />
        </Grid>

        {editStatus ? (
          <div></div>
        ) : (
          <Grid
            item
            xs={12}
            display="flex"
            justifyContent="flex-end"
            pr={{ xs: "20px", sm: "30px" }}
          >
            <Button
              variant="outlined"
              color="error"
              sx={{ marginRight: "10px" }}
              onClick={handleCancel}
            >
              Cancel
            </Button>
            <Button
              type="submit"
              variant="contained"
              color="error"
              onClick={handleSubmit}
            >
              Submit
            </Button>
          </Grid>
        )}
      </Grid>
    </Box>
  );
};

export default ProfileForm;
