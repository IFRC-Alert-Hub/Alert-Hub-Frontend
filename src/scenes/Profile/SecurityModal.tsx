import {
  Box,
  Button,
  InputAdornment,
  TextField,
  Typography,
} from "@mui/material";
import { User } from "../../context/UserContext";
import { useEffect, useState } from "react";
import { useFormik } from "formik";
import * as yup from "yup";
import MessageModal from "./components/MessageModal";
import ModalButton from "./components/ModalButton";
import { useMutation } from "@apollo/client";
import {
  RESET_EMAIL,
  RESET_EMAIL_CONFIRM,
} from "../../API/mutations/authMutations";
import { auth_system } from "../../API/API_Links";

type PropsType = {
  user?: User;
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  setIsVerified: React.Dispatch<React.SetStateAction<boolean>>;
  setEmailToken: React.Dispatch<React.SetStateAction<string>>;
};

const SecurityModal = ({
  user,
  open,
  setOpen,
  setIsVerified,
  setEmailToken,
}: PropsType) => {
  const [isCodeSent, setIsCodeSent] = useState(false);
  const [emailError, setEmailError] = useState(false);
  const [emailRes, setEmailRes] = useState("");
  const [timeLeft, setTimeLeft] = useState(60);

  const [resetEmail] = useMutation(RESET_EMAIL, { client: auth_system });
  const [resetEmailConfirm] = useMutation(RESET_EMAIL_CONFIRM, {
    client: auth_system,
  });

  useEffect(() => {
    if (isCodeSent) {
      const timer = setInterval(() => {
        setTimeLeft((prevTime) => {
          if (prevTime > 0) {
            return prevTime - 1;
          } else {
            setIsCodeSent(false);
            clearInterval(timer);
            return 60;
          }
        });
      }, 1000);

      return () => {
        clearInterval(timer);
      };
    }
  }, [isCodeSent]);

  const formik = useFormik({
    initialValues: {
      verificationCode: "",
    },
    validationSchema: yup.object({
      verificationCode: yup.string().required("Please enter the code"),
    }),
    onSubmit: async () => {
      const res = await resetEmailConfirm({
        variables: { verifyCode: formik.values.verificationCode },
      });
      const { success, token, errors } = res.data.resetEmailConfirm;
      if (success) {
        setOpen(false);
        setIsVerified(true);
        setEmailToken(token);
      } else {
        setEmailError(true);
        setEmailRes(errors.verifyCode);
      }
    },
  });

  const handleClose = () => setOpen(false);
  const handleCodeSent = () => {
    setIsCodeSent(true);
    resetEmail();
  };
  return (
    <MessageModal
      header="Security Verification"
      open={open}
      handleClose={handleClose}
    >
      <Box component="form" onSubmit={formik.handleSubmit}>
        <Typography id="modal-modal-description" color={"#737373"}>
          Click to obtain the verification code. The code will be sent to your
          email address.
        </Typography>
        <Typography id="modal-modal-email" color={"#d30210"}>
          {user?.email}
        </Typography>
        <Box pt={"1rem"}>
          <TextField
            id="verificationCode"
            name="verificationCode"
            size="small"
            placeholder={"Verification Code"}
            value={formik.values.verificationCode}
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            error={
              emailError ||
              (formik.touched.verificationCode &&
                Boolean(formik.errors.verificationCode))
            }
            helperText={
              emailError
                ? emailRes
                : formik.touched.verificationCode &&
                  formik.errors.verificationCode
            }
            fullWidth
            sx={{
              "& .MuiOutlinedInput-root": {
                paddingRight: 0,
              },
            }}
            InputProps={{
              endAdornment: (
                <InputAdornment position="end">
                  <Box borderLeft={"1px solid #ccc"}>
                    <Button
                      variant="text"
                      disableRipple
                      disabled={isCodeSent}
                      sx={{
                        color: "#d30210",
                        p: "0px 10px 0px 10px",
                        minWidth: "50px",
                        textTransform: "capitalize",
                        "&:hover": { opacity: "0.5" },
                      }}
                      onClick={handleCodeSent}
                    >
                      {isCodeSent && timeLeft > 0 ? (
                        <span>{timeLeft}s</span>
                      ) : (
                        <span>Send</span>
                      )}
                    </Button>
                  </Box>
                </InputAdornment>
              ),
            }}
          />
        </Box>
        <Box pt={"2.5rem"}>
          <ModalButton name="Next" />
        </Box>
      </Box>
    </MessageModal>
  );
};

export default SecurityModal;
