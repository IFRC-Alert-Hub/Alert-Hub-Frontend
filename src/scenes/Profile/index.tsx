import { Box, Container, Grid, Typography } from "@mui/material";
import ProfileForm from "./ProfileForm";
import ProfileAvatar from "./ProfileAvatar";
import { useState } from "react";

interface UserType {
  id: string;
  avatar: string;
  firstName: string;
  lastName: string;
  email: string;
  contactNumber: string;
}

const Profile = () => {
  const [user, setUser] = useState<UserType>({
    id: "1",
    avatar: "https://eu.ui-avatars.com/api/?name=M&size=250",
    firstName: "Adi",
    lastName: "Leio",
    email: "email123@email.com",
    contactNumber: "01234567890",
  });
  const [editStatus, setEditStatus] = useState(true);

  return (
    <Container maxWidth={"lg"}>
      <Box margin="30px 0px">
        <Typography
          variant="h2"
          fontWeight={"bold"}
          paddingBottom={"15px"}
          textTransform={"capitalize"}
          letterSpacing={"1.6px"}
        >
          My Profile
        </Typography>

        <Box margin="10px 30px">
          <Grid container>
            <Grid item xs={12} sm={3}>
              <ProfileAvatar
                editStatus={editStatus}
                setEditStatus={setEditStatus}
                user={user}
              />
            </Grid>
            <Grid item xs={12} sm={9}>
              <ProfileForm
                editStatus={editStatus}
                setEditStatus={setEditStatus}
                user={user}
                setUser={setUser}
              />
            </Grid>
          </Grid>
        </Box>
      </Box>
    </Container>
  );
};

export default Profile;
