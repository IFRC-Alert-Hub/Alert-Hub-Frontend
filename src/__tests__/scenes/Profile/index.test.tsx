/* eslint-disable testing-library/no-unnecessary-act */
import { act, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

import Profile from "../../../scenes/Profile";

describe("Proifle", () => {
  it("render Profile component", () => {
    render(<Profile />);

    expect(screen.getByRole("heading")).toBeInTheDocument();
    expect(screen.getByText("Edit Profile")).toBeInTheDocument();
    expect(screen.getByText("Switch")).toBeInTheDocument();
    expect(screen.getByLabelText("First Name")).toBeInTheDocument();
    expect(screen.getByLabelText("Last Name")).toBeInTheDocument();
    expect(screen.getByLabelText("City")).toBeInTheDocument();
    expect(screen.getByLabelText("Country")).toBeInTheDocument();
    expect(screen.getByLabelText("Email Address")).toBeInTheDocument();
  });

  it("click edit button and change profile", async () => {
    render(<Profile />);

    await act(() => {
      userEvent.click(screen.getByText("Edit Profile"));
    });
    expect(screen.getByText("Cancel")).toBeInTheDocument();
    expect(screen.getByText("Submit")).toBeInTheDocument();
    expect(screen.getByText("Switch")).toBeDisabled();

    await act(() => {
      userEvent.type(screen.getByText("First Name"), "test first name");
      userEvent.type(screen.getByText("Last Name"), "test last name");
      userEvent.type(screen.getByText("City"), "test city");
      userEvent.type(screen.getByText("Country"), "test country");
    });
    expect(screen.getByLabelText("First Name")).toHaveValue("test first name");
    expect(screen.getByLabelText("Last Name")).toHaveValue("test last name");
    expect(screen.getByLabelText("City")).toHaveValue("test city");
    expect(screen.getByLabelText("Country")).toHaveValue("test country");
  });

  it("click switch button and show the security modal", () => {
    render(<Profile />);

    act(() => {
      userEvent.click(screen.getByText("Switch"));
    });
    expect(screen.getByText("Security Verification")).toBeInTheDocument();
  });

  it("shows error messages when clicking next button without code", async () => {
    render(<Profile />);

    act(() => {
      userEvent.click(screen.getByText("Switch"));
    });
    act(() => {
      userEvent.click(screen.getByText("Next"));
    });
    const errorContent = await screen.findByText("Please enter the code");
    expect(errorContent).toBeInTheDocument();
  });
});

describe("SubscirptionAlerts Snapshot", () => {
  it("should match DOM snapshot", () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
