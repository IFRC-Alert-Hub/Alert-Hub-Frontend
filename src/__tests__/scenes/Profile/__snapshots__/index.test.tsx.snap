// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscirptionAlerts Snapshot should match DOM snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiBox-root css-14xn2kn"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-8m4rs-MuiTypography-root"
    >
      My Profile
    </h2>
    <div
      className="MuiBox-root css-fr4rql"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3 css-zbaamk-MuiGrid-root"
        >
          <div
            className="MuiBox-root css-oj08aa"
          >
            <div
              className="MuiBox-root css-cet0rr"
            >
              <button
                aria-label="Change your avatar"
                aria-labelledby={null}
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-mui-internal-clone-element={true}
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <div
                  className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar-size css-2s90m6-MuiAvatar-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                    data-testid="PersonIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                    />
                  </svg>
                </div>
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium edit-btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Edit Profile
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 css-173qa7i-MuiGrid-root"
        >
          <form
            autoComplete="off"
            className="form-box MuiBox-root css-0"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="firstName"
                >
                  First Name
                </label>
                <div
                  className="MuiFormControl-root MuiTextField-root form-text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={true}
                      id="firstName"
                      name="firstName"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-ihdtdm"
                      >
                        <span
                          className="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="lastName"
                >
                  Last Name
                </label>
                <div
                  className="MuiFormControl-root MuiTextField-root form-text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={true}
                      id="lastName"
                      name="lastName"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-ihdtdm"
                      >
                        <span
                          className="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="city"
                >
                  City
                </label>
                <div
                  className="MuiFormControl-root MuiTextField-root form-text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={true}
                      id="city"
                      name="city"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-ihdtdm"
                      >
                        <span
                          className="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated form-label css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="country"
                >
                  Country
                </label>
                <div
                  className="MuiFormControl-root MuiTextField-root form-text-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      disabled={true}
                      id="country"
                      name="country"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-ihdtdm"
                      >
                        <span
                          className="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13522eq-MuiGrid-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  htmlFor="email"
                >
                  Email Address
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-agk64j-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={true}
                    id="email"
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                  >
                    <div
                      className="MuiBox-root css-qxb7hr"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1wrvdaj-MuiButtonBase-root-MuiButton-root"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        Switch
                      </button>
                    </div>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-ihdtdm"
                    >
                      <span
                        className="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div />
            </div>
            <div />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
